src/test/java/uk/gov/pay/ledger/transactionsummary/dao/TransactionSummaryDaoIT.java

```diff
@@ -299,4 +303,228 @@ public void report_volume_total_amount_and_average_amount_for_all_payments() {
+                .insert(rule.getJdbi());
+
+        var queryWithoutMoto = new TransactionSummaryStatisticQuery()
+                .withAccountId(gatewayAccountId)
+                .withFromDate("2019-09-29T10:00:00Z");
+
+        var queryWithMoto = new TransactionSummaryStatisticQuery()
+                .withAccountId(gatewayAccountId)
+                .withMoto(true)
+                .withFromDate("2019-09-29T10:00:00Z");
+
+        TransactionsStatisticsResult paymentsStatistics = transactionSummaryDao.getTransactionSummaryStatistics(queryWithoutMoto, TransactionType.PAYMENT);
+        assertThat(paymentsStatistics.getCount(), is(2L));
+        assertThat(paymentsStatistics.getGrossAmount(), is(5000L));
+
+        TransactionsStatisticsResult refundsStatistics = transactionSummaryDao.getTransactionSummaryStatistics(queryWithoutMoto, TransactionType.REFUND);
+        assertThat(refundsStatistics.getCount(), is(1L));
+        assertThat(refundsStatistics.getGrossAmount(), is(1000L));
+
+        TransactionsStatisticsResult motoPaymentsStatistics = transactionSummaryDao.getTransactionSummaryStatistics(queryWithMoto, TransactionType.PAYMENT);
+        assertThat(motoPaymentsStatistics.getCount(), is(1L));
+        assertThat(motoPaymentsStatistics.getGrossAmount(), is(1000L));
+    }
+
+    @Test
+    public void shouldSumAmountWhenTotalAmountIsNull() {
+        String gatewayAccountId = RandomStringUtils.randomAlphanumeric(10);
+        aTransactionSummaryFixture()
+                .withGatewayAccountId(gatewayAccountId)
+                .withType(TransactionType.PAYMENT)
+                .withTransactionDate(LocalDate.parse("2019-09-30"))
+                .withAmount(4000L)
+                .withNoOfTransactions(4L)
+                .withState(TransactionState.SUCCESS)
+                .insert(rule.getJdbi());
+        aTransactionSummaryFixture()
+                .withGatewayAccountId(gatewayAccountId)
+                .withType(TransactionType.REFUND)
+                .withTransactionDate(LocalDate.parse("2019-10-01"))
+                .withAmount(1000L)
+                .withState(TransactionState.SUCCESS)
+                .withNoOfTransactions(1L)
+                .insert(rule.getJdbi());
+
+        var query = new TransactionSummaryStatisticQuery()
+                .withAccountId(gatewayAccountId)
+                .withFromDate("2019-09-29T10:00:00Z");
+
+        TransactionsStatisticsResult paymentsStatistics = transactionSummaryDao.getTransactionSummaryStatistics(query, TransactionType.PAYMENT);
+        assertThat(paymentsStatistics.getCount(), is(4L));
+        assertThat(paymentsStatistics.getGrossAmount(), is(4000L));
+
+        TransactionsStatisticsResult refundsStatistics = transactionSummaryDao.getTransactionSummaryStatistics(query, TransactionType.REFUND);
+        assertThat(refundsStatistics.getCount(), is(1L));
+        assertThat(refundsStatistics.getGrossAmount(), is(1000L));
+    }
```

Comment:
This could do with a test where we want just one specific day, eg:
"&from_date=2019-09-29T00:00:00.000Z&to_date=2019-09-29T23:59:59.000Z"

Reply:
Agreed - the code converts from ZonedDateTime to LocalDate so with <= will return all transactions on say 29th September 2019.